extensions/wikibase/tests/src/org/openrefine/wikibase/commands/PreviewWikibaseSchemaCommandTest.java

```diff
@@ -110,4 +120,33 @@ public void testInvalidManifest() throws IOException, ServletException {
         assertEquals(writer.toString(),
                 "{\"code\":\"error\",\"message\":\"Wikibase manifest could not be parsed. Error message: invalid manifest format\"}");
     }
+
+    @Test
+    public void testWarningData() throws Exception {
+
+        String schemaJson = jsonFromFile("schema/warning_data_test.json");
+        String manifestJson = jsonFromFile("manifest/wikidata-manifest-v1.0.json");
+        when(request.getParameter("project")).thenReturn(String.valueOf(project.id));
+        when(request.getParameter("schema")).thenReturn(schemaJson);
+        when(request.getParameter("manifest")).thenReturn(manifestJson);
+
+        command.doPost(request, response);
+
+        ObjectNode response = ParsingUtilities.evaluateJsonStringToObjectNode(writer.toString());
+        ArrayNode issues = (ArrayNode) response.get("warnings");
+        for (JsonNode node : issues) {
+            // Read aggregationId
+            String aggregationId = node.get("aggregationId").asText();
+            // Check for nested property properties/missing_property_entity/label
+            JsonNode addedPropertyLabel = node.path("properties").path("added_property_entity").path("label");
+            JsonNode itemEntityLabel = node.path("properties").path("item_entity").path("label");
+
+            if (aggregationId.equals("existing-item-requires-certain-other-statement-with-suggested-value_P633P17")) {
```

Comment:
That's an `if` condition - note that you are not asserting that this warning id is emitted at all.

Reply:
Fixed